File: src/tractor/CardCollection.java
package tractor;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Iterator;
import java.util.List;
import java.util.Random;

public class CardCollection implements Iterable<Card> {
	protected List<Card> cards;

	public CardCollection() {
		cards = new ArrayList<Card>();
	}

	public void addCards(CardCollection cc) {
		for (Card i : cc) {
			cards.add(i);
		}
	}

	public void addCard(Card card) {
		cards.add(card);
	}

	public int countSuit(int suit) {
		int count = 0;
		for (Card i : cards) {
			if (i.getSuit() == suit) {
				count++;
			}
		}
		return count;
	}

	public Card get(int i) {
		return cards.get(i);
	}

	public int getCount(Card c) {
		return Collections.frequency(cards, c);
	}

	public int getCardCount() {
		return cards.size();
	}

	public boolean isEmpty() {
		return cards.isEmpty();
	}

	@Override
	public Iterator<Card> iterator() {
		return cards.iterator();
	}

	public boolean remove(Card i) {
		return cards.remove(i);
	}

	public Card remove(int i) {
		return cards.remove(i);
	}

	public void removeAll() {
		cards.removeAll(cards);
	}

	public void removeCards(CardCollection cards) {
		for (Card i : cards) {
			this.cards.remove(i);
		}
	}

	public void shuffle() {
		long seed = TractorGame.TESTING ? 1445123063075L : System.currentTimeMillis();
		Collections.shuffle(cards, new Random(seed));
	}

	public int size() {
		return cards.size();
	}

	public void sort() {
		Collections.sort(cards);
	}
}

File: src/python/CardCollection.py
import random
from TestingSuite import TESTING

class CardCollection:
  def __iter__(self):
    return self
    
  def __next__(self):
    if self.index == len(self.cards):
      self.index = 0
      raise StopIteration
    else:
      self.index += 1
      return self.cards[self.index - 1]
      
  def __init__(self):
    self.index = 0
    self.cards = []

  def addCards(self, cc): 
    for i in cc:
      self.cards.append(i)

  def addCard(self, card):
    self.cards.append(card)

  def countSuit(self, suit):
    count = 0
    for i in self.cards:
      if i.getSuit() == suit:
        count += 1
    return count

  def get(self, i):
    return self.cards[i]

  def getCount(self, c):
    return sum(c == card for card in self.cards)

  def getCardCount(self):
    return len(self.cards)

  def isEmpty(self):
    return len(self.cards) == 0

  def remove(self, i):
    if type(i) != int:
      return self.cards.remove(i)
    else:
      return self.cards.pop(i)

  def removeAll(self):
    self.cards.clear()

  def removeCards(self, cards):
    for i in cards.cards:
      self.cards.remove(i)
      
  def shuffle(self):
    if TESTING:
      random.seed(0)
    else:
      random.seed()
    random.shuffle(self.cards)

  def size(self):
    return len(self.cards)
    
  def sort(self): 
    self.cards.sort()